app/controllers/authors_controller.rb

```diff
@@ -0,0 +1,41 @@
+class AuthorsController < ApplicationController
+  def index
+    authors=Author.all
+    render json: authors, status: 200
+  end
+
+  def show
+    begin
+      author=Author.find(params[:id])
+      if author
+        render json: author, status: 200
+      end
+    rescue StandardError => e
+      render json:{
+        error:e
+      }
+    end
+  end
+
+  def create
+    begin
+      author=Author.new(
```

Comment:
find or create the user using email

Reply:
Find user using email functionality implemented, please check latest commit. email was used to create the user in the earlier commit.